Nodejs/Product/Nodejs/SharedProject/ProjectNode.cs

```diff
@@ -6026,57 +6028,44 @@ private void SetupProjectGlobalPropertiesThatAllProjectSystemsMustSet()
         /// </remarks>
         private bool TryBeginBuild(bool designTime, bool requiresUIThread = false)
         {
-            var releaseUIThread = false;
-
-            try
+            // If the SVsBuildManagerAccessor service is absent, we're not running within Visual Studio.
+            if (this.buildManagerAccessor != null)
             {
-                // If the SVsBuildManagerAccessor service is absent, we're not running within Visual Studio.
-                if (this.buildManagerAccessor != null)
+                if (requiresUIThread)
                 {
-                    if (requiresUIThread)
+                    var result = this.buildManagerAccessor.ClaimUIThreadForBuild();
+                    if (result < 0)
                     {
-                        var result = this.buildManagerAccessor.ClaimUIThreadForBuild();
-                        if (result < 0)
-                        {
-                            // Not allowed to claim the UI thread right now. Try again later.
-                            return false;
-                        }
-
-                        releaseUIThread = true; // assume we need to release this immediately until we get through the whole gauntlet.
+                        // Not allowed to claim the UI thread right now. Try again later.
+                        return false;
                     }
+                }
 
-                    if (designTime)
+                if (designTime)
+                {
+                    var result = this.buildManagerAccessor.BeginDesignTimeBuild();
```

Comment:
Can this throw an exception? If so, I think the UI thread will be blocked and never released.

Reply:
As above, I believe it's technically possible, but I'm not convinced we should attempt to recover if it does.